### DISCUSSION TITLE
Is it possible to specify multiple targets from a single source?

### DISCUSSION THREAD

**Discussion by Stealthmate** [ruby/rake#436] [Q&A]
Basically I have a binary `foo` that generates `1.bar`, `2.bar` and `3.bar`. So I wrote a rule

```ruby
rule(/[1-3]\.bar/ => [ 'foo' ] do |t|
  sh "foo"
end
```

But if I `rake` something that needs all 3 outputs, `foo` gets executed 3 times. My problem is that `foo` is kind of slow, so executing it 3 times is annoying. Is there a way to make rake only run it once? If not, would it be possible to implement it?

I think the relevante Make concept is [Grouped Targets](https://www.gnu.org/software/make/manual/html_node/Multiple-Targets.html).

**Comment by jgomo3** [comment]
I think that if the action is defined for `foo` instead for the rule, then it will be ran once.

```ruby
task :foo do |t|
  sh "foo"
end

rule(/[1-3]\.bar/ => [ 'foo' ]
```
